python/lsst/analysis/ap/nb_utils.py

```diff
@@ -70,3 +78,203 @@ def make_simbad_link(ra, dec, radius_arcsec=3.0):
+
+        index, sep, _ = match_coordinates_sky(coords1, coords2)
+        indices[(visit, detector)] = index
+        seps[(visit, detector)] = sep
+        gs1['xmatch_dist_arcsec'] = sep.to_value(u.arcsecond)
+        gs1['src2_diaSourceId'] = gs2['diaSourceId'].values.astype(np.int64)[index]
+
+        # Set the match ID to 0 if the distance is above threshold
+        gs1.loc[(gs1['xmatch_dist_arcsec'] > matchRadius),
+                ['src2_diaSourceId']] = 0
+
+        # get the DiaSources in dataset 2 not matched to something in dataset 1
+        uniqueid2 = set(gs2['diaSourceId']) - set(gs1['src2_diaSourceId'])
+        unique2 = gs2[gs2['diaSourceId'].isin(uniqueid2)]
+
+        unique1 = gs1[(gs1['src2_diaSourceId'] == 0)]
+
+        withmatch = gs1[(gs1['src2_diaSourceId'] > 0)]
+
+        src1.append(unique1)
+        src2.append(unique2)
+        matches.append(withmatch)
+
+    # Out of the loop, concatenate everything together.
+    unique1 = pd.concat(src1)
+    unique2 = pd.concat(src2)
+    matched = pd.concat(matches)
+
+    # Make cutouts if they don't already exist; optionally display to screen
+    config = pisco.PlotImageSubtractionCutoutsConfig()
+    config.url_root = ''
+    config.add_metadata = add_cutout_metadata
+    config.diff_image_type = diffimg_type1
+
+    if cutoutPath is None:
+        cutoutPath = os.environ['PWD']
+        logging.warning('cutoutPath not supplied; defaulting to '+cutoutPath)
+
+    piscot = pisco.PlotImageSubtractionCutoutsTask(output_path=cutoutPath,
+                                                   config=config)
+
+    print("{} matched sources; {} unique to set 1; {} unique to set 2.".format(
+        len(matched), len(unique1), len(unique2)))
+
+    # First figure out which cutouts already exist at the output path
+    unique1['pathexists'] = False
+    for i in range(len(unique1)):
+        dId = unique1.iloc[i]['diaSourceId']
+        idx = unique1.index[i]
+        unique1.at[idx, 'pathexists'] = os.path.exists(
+            os.path.join(
+                cutoutPath,
+                f"images/{int(dId // chunk_size) * chunk_size}/{dId}.png"))
+    pathchk1 = unique1.loc[~unique1['pathexists']]
+
+    unique2['pathexists'] = False
+    for i in range(len(unique2)):
+        dId = unique2.iloc[i]['diaSourceId']
+        idx = unique2.index[i]
+        unique2.at[idx, 'pathexists'] = os.path.exists(
+            os.path.join(
+                cutoutPath,
+                f"images/{int(dId // chunk_size) * chunk_size}/{dId}.png"))
+    pathchk2 = unique2.loc[~unique2['pathexists']]
+
+    # Only write those that don't exist yet
+    piscot.write_images(pathchk1, butler=butler1, njobs=njobs)
+    piscot.config.diff_image_type = diffimg_type2
+    piscot.write_images(pathchk2, butler=butler2, njobs=njobs)
+
+    if displayCutouts:
+        for isrc in unique1.itertuples():
+            chunked = int((isrc.diaSourceId // chunk_size) * chunk_size)
```

Comment:
Suggest removing the chunking, as above comment.

Reply:
See other comment about using the managed path.